docs/proposals/compute-resource-metrics-api.md

```diff
@@ -61,13 +77,53 @@ kube-ui-v1-nd7in           0.07 cores  130 MB
 A second user will be the scheduler. To assign pods to nodes efficiently, the
 scheduler needs to know the current free resources on each node.
 
-## Proposed endpoints
+The Kubelet API will be used by heapster to provide metrics at the
+cluster-level. The Kubelet API will also be useful for debugging individual
+nodes, and stand-alone kubelets.
 
-	/api/v1/namespaces/myns/podMetrics/mypod
-	/api/v1/nodeMetrics/myNode
+## Proposed endpoints
 
-The derived metrics include the mean, max and a few percentiles of the list of
-values.
+The metrics API will be its own [API group](api-group.md), and is shared by the
+kubelet and cluster API.  All endpoints are GET endpoints, rooted at
+`/apis/metrics/v1alpha1/`
+
+- `/` - discovery endpoint; type resource list
+- `/nodes` - host metrics; type `[]metrics.Node`
+  - `/nodes/localhost` - When served by kubelet, the only node provided is
+    `localhost`; type metrics.Node
+  - `/nodes/{node}` - metrics for a specific node
+- `/derivedNodes` - host metrics; type `[]metrics.DerivedNode`
+  - `/derivedNodes/localhost` - When served by kubelet, the only node provided is
+    `localhost`; type metrics.Node
+  - `/nodes/{node}` - derived metrics for a specific node
+- `/pods` - All pod metrics across all namespaces; type `[]metrics.Pod`
+- `/derivedPods` - All pod metrics across all namespaces; type `[]metrics.Pod`
+- `/namespaces/{namespace}/pods` - All pod metrics within namespace; type
+  `[]metrics.Pod`
+  - `/namespaces/{namespace}/pods/{pod}` - metrics for specific pod
+- `/namespaces/{namespace}/derivedPods` - All derived pod metrics within
+  namespace; type `[]metrics.DerivedPod`
+  - `/namespaces/{namespace}/derivedPods/{pod}` - derived metrics for specific
+  pod
+- Unsupported paths return status not found (404)
+  - `/namespaces/`
+  - `/namespaces/{namespace}`
+
+Additionally, all endpoints (except root discovery endpoint) support the
+following optional query parameters:
+
+- `start` - start time to return metrics from; type json encoded `time.Time`
+- `end` - end time to return metrics to; type json encoded `time.Time`
+- `step` - the time step between each stats sample; type int (seconds), default
+  10s, must be a multiple of 10s
+
+As well as the common query parameters:
+
+- `pretty` - pretty print the response
```

Comment:
Can you educate me with an example response with this query parameter @timstclair ?